### PR TITLE
Support collection format in modular

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1984

When I implement the paging feature in modular I found the collect format is not supported and this would cause the compile error. So I need to fix it before [this pr](https://github.com/Azure/autorest.typescript/pull/1978).

- Enable the collection format test cases in cadl-ranch
- Handle the operation name conflicts

### CODE REVIEW

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 1]
```diff
@@ -1318,9 +1319,38 @@ function emitOperationGroups(
       operations: clientOperations
     });
   }
+  resolveConflictIfExist(operationGroups);
   return operationGroups;
 }
 
+function resolveConflictIfExist(operationGroups: OperationGroup[]) {
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1326]: Detect if there is name conflict cross operation groups, if yes, add operation group name prefix.

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 2]
```diff
@@ -407,6 +417,22 @@ function getParameterMap(
   throw new Error(`Parameter ${param.clientName} is not supported`);
 }
 
+function getCollectionFormat(param: Parameter) {
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:420]: Build the param with collection format helper function.

File: packages/typespec-ts/src/modular/buildCodeModel.ts (hunk 1 above)
Comment (qiaozha) [reply packages/typespec-ts/src/modular/buildCodeModel.ts:1326]: Can you add this check for RLC as well ? thx.

Comment (MaryGao) [reply packages/typespec-ts/src/modular/buildCodeModel.ts:1326]: Done.

File: packages/typespec-ts/test/modularIntegration/generated/parameters/collection-format/src/models/models.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Copyright (c) Microsoft Corporation.
```
[the remark this answers is not in the document]
Comment (MaryGao) [reply packages/typespec-ts/test/modularIntegration/generated/parameters/collection-format/src/models/models.ts]: Done

**Review by qiaozha — APPROVED** [review]

**Pull request merged**